website/server/models/user/schema.js

```diff
@@ -24,7 +24,21 @@ let schema = new Schema({
     local: {
       email: {
         type: String,
-        validate: [validator.isEmail, shared.i18n.t('invalidEmail')],
+        validate: [{
+          validator: validator.isEmail,
+          msg: shared.i18n.t('invalidEmail'),
+        }, {
+          validator: function notEmailDomains (email) {
+            let domains = ['habitica\\.com', 'habitrpg\\.com'];
```

Comment:
I'd put these as a const near the top of the file:

```js
const INVALID_DOMAINS = ['habitica.com', 'habitrpg.com'];
```

And then in the validator, I'd just do this:

```js
validator (email) {
  return INVALID_DOMAINS.every(domain => email.indexOf(domain) === -1);
}
```